code/game/machinery/_machinery.dm

```diff
@@ -159,10 +159,18 @@
 	fire = 50
 	acid = 70
 
-/obj/machinery/Initialize(mapload)
+/obj/machinery/Initialize(mapload, obj/structure/frame/machine/frame_with_parts = null, mob/living/builder = null, obj/item/building_tool = null)
 	. = ..()
 	SSmachines.register_machine(src)
 
+	if(frame_with_parts && istype(frame_with_parts, /obj/structure/frame/machine))
```

Comment:
As stated above frame_with_parts is only used by machine frame so this null and istype() check adds init time across all machines and does nothing so it's a wast

Reply:
So went want to keep the current process where machines that are manually built first put in all the default parts for a circuitboard, delete all those parts, look at the parts for the frame, and then add the parts from the frame? Additionally, the logic for vendors is half-broken due to it assuming that its circuit is going to be an object, when the early call to parent obj/machinery/Initialization will finish most of the part-dependent initialization (including, it seems, pricing and stocking).

Reply:
> So we want to keep the current process where machines that are manually built first put in all the default parts for a circuitboard, delete all those parts, look at the parts for the frame, and then add the parts from the frame?

A better way is to stop the machine from applying the default parts just for machine frames by adding a boolean var to this line

https://github.com/tgstation/tgstation/blob/5a2231be3b7a14feb74e6553309b0ec07d559108/code/game/machinery/_machinery.dm#L174-L176

And rewrite it to something like

```
/obj/machinery/Initializize(mapliad, apply_default_parts)
..
 if(ispath(circuit, /obj/item/circuitboard) && apply_default_parts) 
```

We introduce our own parameter as you can see `apply_default_parts`. Then you can remove the part of deleting default parts which is done by the machine frame making it much more cleaner.

Also i don't think your code  even works currently because you are still applying the default parts ayway in the above mentioned line just after you moved in the machine frame parts from your code

Reply:
I've creeped the scope on this as a matter of fact; I branched another one to do machine init and I'm gonna rollback and redo what I have here.